### ISSUE TITLE
FGS "Use logit mode if computing loss w.r.t. output scores rather than softmax"

### ISSUE THREAD

**Issue by ysharma1126** [facebookresearch/adversarial_image_defenses#4]
So mode needs to be set to FGSMode.LOGIT, if I want to use the logits? If mode is 'None', the cross-entropy loss is used, if mode is 'FGSMode.CARLINI', the output is updated, then the FGSMode.LOGIT loss is used.

Not entirely sure what's going on. If I wanted to use the logits, and use the xent loss, what mode should I set?

**Comment by lvdmaaten** [comment]
You cannot use a cross-entropy loss on logits: a cross-entropy loss operates on probabilities (obtained via, for instance, a softmax operation), not on unnormalized scores. The different mode behaviors are defined [here](https://github.com/facebookresearch/adversarial_image_defenses/blob/84e47bce290d4f4eb0d24e92ea77f1bc9357becd/adversarial/lib/adversary.py#L71-L79).
